### ISSUE TITLE
Extracting list of requirements from a conanfile or Conan recipe

### ISSUE THREAD

**Issue by JPGygax68** [conan-io/conan#1627]
Is it possible using commands only? 

If not, handling conanfiles would be easy, but how to go about it for recipes? Would it be a good idea to load the recipe module via `importlib` and try to instantiate the class, or would it be sufficient to just extract the statically assigned value of `requires`?

**Comment by memsharded** [comment]
Please not that requires is not static information. It might be dynamically built, being different for different configurations, OSs, or using different options. So you have a dependency graph for each configuration.

You can try the ``conan info`` command, you can filter fields, and also packages:

```bash
$ conan info Pkg/0.1@user/testing --only=requires --package_filter=Pkg/0.1*
```

**Comment by ovidiub13** [comment]
I think he means to use the values from that `requires` in another script.

**Comment by JPGygax68** [comment]
Quite right. I'll just have to parse the output of that command. Tedious,
but much safer.

One could wish for some kind of machine-readable output flag, but that
won't be necessary in this case.

Am 18.08.2017 07:41 schrieb "Ovidiu-Florin BOGDAN" <notifications@github.com
>:

I think he means to use the values from that requires in another script.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/conan-io/conan/issues/1627#issuecomment-323264640>, or mute
the thread

**Comment by JPGygax68** [comment]
Please disregard the comment about machine-readable. The output is perfectly parsable.

**Comment by memsharded** [comment]
Ok, @JPGygax68 good to know. Just a note: we have internally implemented a python API to call conan commands, for better integration with other python tools. So if you are using python, that might be useful.

**Comment by JPGygax68** [comment]
Thanks James, for the info. I take it that that API, or parts of it, will be made available in a future release of Conan?

**Comment by memsharded** [comment]
Exactly that: it is already available in 0.25, but we didn't document it yet, waiting a bit to stabilize it. But normal operation should be relatively stable, so if you are in need of this feature, I would suggest to give it a try, the map to conan commands is very straightforward, so looking to the source code could be enough.
